receiver.go

```diff
@@ -541,36 +540,34 @@ func (r *Receiver) mux() {
 		// unblock any in flight message dispositions
 		r.inFlight.clear(r.l.err)
 
-		// unblock any pending drain requests
-		if r.manualCreditor != nil {
-			r.manualCreditor.EndDrain()
+		if !r.autoSendFlow {
+			// unblock any pending drain requests
+			r.creditor.EndDrain()
 		}
 	}, func(fr frames.PerformTransfer) {
 		_ = r.muxReceive(fr)
 	})
 
 	for {
-		switch {
-		case r.manualCreditor != nil:
-			drain, credits := r.manualCreditor.FlowBits(r.l.availableCredit)
-
-			if drain || credits > 0 {
-				debug.Log(1, "receiver (manual): source: %s, inflight: %d, credit: %d, creditsToAdd: %d, drain: %v, deliveryCount: %d, messages: %d, unsettled: %d, maxCredit : %d, settleMode: %s",
-					r.l.source.Address, r.inFlight.len(), r.l.availableCredit, credits, drain, r.l.deliveryCount, len(r.messages), r.countUnsettled(), r.maxCredit, r.l.receiverSettleMode.String())
-
-				// send a flow frame.
-				r.l.err = r.muxFlow(credits, drain)
-			}
-
-		// if receiver && half maxCredits have been processed, send more credits
-		case r.l.availableCredit+uint32(r.countUnsettled()) <= r.maxCredit/2:
+		// if half maxCredits have been processed and auto-flow is enabled, send more credits
+		if available := r.l.availableCredit + uint32(r.countUnsettled()); available <= r.maxCredit/2 && r.autoSendFlow {
 			debug.Log(1, "receiver (half): source: %s, inflight: %d, credit: %d, deliveryCount: %d, messages: %d, unsettled: %d, maxCredit : %d, settleMode: %s", r.l.source.Address, r.inFlight.len(), r.l.availableCredit, r.l.deliveryCount, len(r.messages), r.countUnsettled(), r.maxCredit, r.l.receiverSettleMode.String())
+			r.l.err = r.creditor.IssueCredit(r.maxCredit-available, r)
+		} else if r.l.availableCredit == 0 {
+			debug.Log(1, "receiver (pause): inflight: %d, credit: %d, deliveryCount: %d, messages: %d, unsettled: %d, maxCredit : %d, settleMode: %s", r.inFlight.len(), r.l.availableCredit, r.l.deliveryCount, len(r.messages), r.countUnsettled(), r.maxCredit, r.l.receiverSettleMode.String())
+		}
 
-			linkCredit := r.maxCredit - uint32(r.countUnsettled())
-			r.l.err = r.muxFlow(linkCredit, false)
+		if r.l.err != nil {
+			return
+		}
 
-		case r.l.availableCredit == 0:
-			debug.Log(1, "receiver (pause): inflight: %d, credit: %d, deliveryCount: %d, messages: %d, unsettled: %d, maxCredit : %d, settleMode: %s", r.inFlight.len(), r.l.availableCredit, r.l.deliveryCount, len(r.messages), r.countUnsettled(), r.maxCredit, r.l.receiverSettleMode.String())
+		drain, credits := r.creditor.FlowBits(r.l.availableCredit)
+		if drain || credits > 0 {
+			debug.Log(1, "receiver: source: %s, inflight: %d, credit: %d, creditsToAdd: %d, drain: %v, deliveryCount: %d, messages: %d, unsettled: %d, maxCredit : %d, settleMode: %s",
+				r.l.source.Address, r.inFlight.len(), r.l.availableCredit, credits, drain, r.l.deliveryCount, len(r.messages), r.countUnsettled(), r.maxCredit, r.l.receiverSettleMode.String())
+
```

Comment:
Nothing wrong with this but now that I look at it more it looks like 'availableCredit' should really be called 'issuedCredit'.

Reply:
Agreed that `available` isn't correct.  I'm thinking that `pendingCredit` is more descriptive and is what I've been testing locally.

Reply:
fixed